mesonbuild/cmake/interpreter.py

```diff
@@ -714,7 +714,7 @@ def ensure_absolute(x: Path) -> Path:
             # targets, etc. This reduces the chance of misdetecting input files
             # as outputs from other targets.
             # See https://github.com/mesonbuild/meson/issues/6632
-            if not raw.is_absolute() and (self.current_src_dir / raw).exists():
+            if not raw.is_absolute() and (self.current_src_dir / raw).exists() and (self.current_src_dir / raw).is_file():
```

Comment:
```suggestion
            if not raw.is_absolute() and (self.current_src_dir / raw).is_file():
```

We don't need to check both.

Reply:
ping?

Reply:
Forgot about that one...